### PR TITLE
Fix project directory has spaces lead to compile error

### PR DESCRIPTION
**Pull request by kongshanshan33**
### Steps to Reproduce:

1. I use 'flutter create -t module module_test' command in directory 'flutter test' 

2. Opened this project using android studio

3. FAILURE: Build failed with an exception

### Logs：

Could not determine the dependencies of task ':app:lintVitalRelease'.
> Could not resolve all task dependencies for configuration ':app:releaseRuntimeClasspath'.
   > Could not resolve project :flutter.
     Required by:
         project :app
      > Unable to find a matching configuration of project :flutter: None of the consumable configurations have attributes.

### CODE REVIEW

**Comment by kongshanshan33** [comment]
I signed it!

**Comment by Hixie** [comment]
We should add a test for this. In the past we've tested this kind of thing by having our entire CI infrastructure rename the directory to add a space in the path; maybe we broke than when moving to Cirrus?

cc @gspencergoog @goderbauer

**Comment by gspencergoog** [comment]
No, Cirrus still installs into "flutter sdk" (with a space).

**Comment by dnfield** [comment]
This is specific to the Add2App template stuff - I don't think we have as many tests around that

/cc @matthew-carroll

File: packages/flutter_tools/templates/module/android/library/include_flutter.groovy.copy.tmpl [hunk 1]
```diff
@@ -1,6 +1,6 @@
 // Generated file. Do not edit.
 
-def scriptFile = getClass().protectionDomain.codeSource.location.path
+def scriptFile = getClass().protectionDomain.codeSource.location.toURI()
```
Comment (aam) [inline packages/flutter_tools/templates/module/android/library/include_flutter.groovy.copy.tmpl]: Is this change Windows-friendly?

Comment (aam) [reply packages/flutter_tools/templates/module/android/library/include_flutter.groovy.copy.tmpl]: Looks like it is.

**Review by aam — APPROVED** [review]

**Comment by aam** [comment]
Created https://github.com/flutter/flutter/issues/26304 to track setting up Add2App test infrastructure.

**Comment by aam** [comment]
Thank you for your contribution @kongshanshan33 !

**Pull request merged**